mulighetsrommet-api/src/main/kotlin/no/nav/mulighetsrommet/api/services/TiltaksgjennomforingService.kt

```diff
@@ -39,20 +47,26 @@ class TiltaksgjennomforingService(
         if (avtale.sluttDato.isBefore(currentDate)) {
             return Either.Left(BadRequest("Avtalens sluttdato har passert"))
         }
+        virksomhetService.getOrSyncVirksomhet(request.arrangorOrganisasjonsnummer)
 
         val previousAnsvarlig = tiltaksgjennomforingRepository.get(request.id)?.ansvarlig?.navident
 
         return request.toDbo()
-            .onRight { virksomhetService.hentEnhet(it.arrangorOrganisasjonsnummer) }
-            .onRight { tiltaksgjennomforingRepository.upsert(it) }
-            .onRight {
-                if (navIdent != request.ansvarlig && request.ansvarlig != previousAnsvarlig) {
-                    sattSomAnsvarligNotification(it.navn, request.ansvarlig)
+            .map { dbo ->
+                db.transaction { tx ->
+                    tiltaksgjennomforingRepository.upsert(dbo, tx)
+                    utkastRepository.delete(dbo.id, tx)
+                    if (navIdent != request.ansvarlig && request.ansvarlig != previousAnsvarlig) {
+                        sattSomAnsvarligNotification(dbo.navn, request.ansvarlig, tx)
+                    }
+
+                    val dto = tiltaksgjennomforingRepository.get(request.id, tx)!!
+
+                    tiltaksgjennomforingKafkaProducer.publish(TiltaksgjennomforingDto.from(dto))
+                    dto
                 }
             }
-            .map { tiltaksgjennomforingRepository.get(request.id)!! }
             .onRight { sanityTiltaksgjennomforingService.opprettSanityTiltaksgjennomforing(it) }
```

Comment:
I stedet for å lage en lytter på kafka som gjør dette i etterkant lot jeg den bare være her enn så lenge 🤷

Reply:
Gir mening! Og dette blir ifm. publisering til migrering-topic, om jeg husker riktig?